cli/dcos-spark/submit_builder_test.go

```diff
@@ -49,6 +51,40 @@ func (suite *CliTestSuite) TestCleanUpSubmitArgs() {
 	}
 }
 
+func (suite *CliTestSuite) TestCleanUpSubmitArgsWithMulitpleValues() {
+	_, args := sparkSubmitArgSetup()
+	inputArgs := "--conf spark.driver.extraJavaOptions='-XX:+PrintGC -Dparam1=val1 -Dparam2=val2' main.py 100"
+	expected := "--conf=spark.driver.extraJavaOptions=-XX:+PrintGC -Dparam1=val1 -Dparam2=val2"
```

Comment:
Here is the problem. The quoting is stripped from the argument. In the CI, it is most likely the second `-XX` that is failing, but I would expect the `-D` to fail here aswell.

Reply:
Do confirm, we may want to include tests that call `buildSubmitJson` with the relevant arguments too.

Reply:
yeah i've tested it on CI where even just setting `extraJavaOptions='val1'` results in error `Cannot read file val1` or something like that. 

I think this is Python doing its own kind of parsing before getting to Go parsing and that causes the issue.

In commit [7c294c7](https://github.com/mesosphere/spark-build/pull/388/commits/7c294c7e19bd6249e5f01bbb7723c4a0c884dd43) I added that same config passed to `buildSubmitJson` and it still works.

Reply:
BTW, I might have missed it in the cleanupSubmitArgs logic: when do the single quotes get stripped off?

Reply:
Not in our logic, it happens in `shellwords.parse()` 
```
args, err := shellwords.Parse(argsStr)
```